src/sentry/integrations/issue_alert_image_builder.py

```diff
@@ -35,30 +36,49 @@ def __init__(self, group: Group, provider: ExternalProviderEnum) -> None:
 
     def get_image_url(self) -> str | None:
         try:
-            image_url = None
             metrics.incr("chartcuterie.issue_alert.attempt", tags=self.tags)
-            if self.group.issue_type == PerformanceP95EndpointRegressionGroupType:
-                image_url = self._get_endpoint_regression_image_url()
-            elif self.group.issue_type == ProfileFunctionRegressionType:
-                image_url = self._get_function_regression_image_url()
-
+            image_url = cache.get(self.cache_key)
```

Comment:
Can we move the cache fetching into the `generate_chart` function?

Though I'm guessing you want to do this early before we have to query Snuba again for all the data right?

In that case what I might suggest is adjusting the `generate_chart` so that `data` can be a function that generates the data if needed (ie, the cache lookup is a miss)

Alternatively, let's do all the caching outside the `generate_chart`. Right now we've coupled implementation details of `generate_chart` to implementation details of its caller.

Reply:
Yeah, here I'm trying to query `event-stats` once. I'll take the cache logic out since it's easier 😅

Reply:
👍